source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -180,11 +282,69 @@ void TextureBaker::writeBakedDocument(NodePtr shader, const FilePath& filename)
     writeToXmlFile(bakedTextureDoc, filename);
 }
 
-FilePath TextureBaker::generateTextureFilename(OutputPtr output)
+FilePath TextureBaker::generateTextureFilename(OutputPtr output, const string& srName, const string& udim)
 {
     string outputName = createValidName(output->getNamePath());
-    string udimSuffix = _udim.empty() ? EMPTY_STRING : "_" + _udim;
-    return FilePath(outputName + "_baked" + udimSuffix + "." + _extension);
+    string srSegment = srName.empty() ? EMPTY_STRING : "_" + srName;
+    string udimSuffix = udim.empty() ? EMPTY_STRING : "_" + udim;
+
+    return FilePath(outputName + srSegment + "_baked" + udimSuffix + "." + _extension);
+}
+
+void TextureBaker::bakeAllShaders(DocumentPtr& doc, string file, bool hdr, int texres)
+{
+    TextureBakerPtr baker = TextureBaker::create(texres, texres, hdr ? Image::BaseType::FLOAT : Image::BaseType::UINT8);
+    FileSearchPath filename = initFileSearchPath();
+    GenContext genContext = initGenContext();
+    genContext.registerSourceCodeSearchPath(filename);
+    StringVec renderablePaths = getRenderablePaths(doc);
+    StringResolverPtr resolver = StringResolver::create();
+    for (const auto& renderablePath : renderablePaths)
+    {
+        ElementPtr elem = doc->getDescendant(renderablePath);
+        TypedElementPtr typedElem = elem ? elem->asA<TypedElement>() : nullptr;
+        if (!typedElem || elem->getCategory() != "shaderref")
```

Comment:
There's a more compact way to accomplish this that's more commonly used in MaterialX:

```
ShaderRefPtr sr = typedElem ? typedElem ->asA<ShaderRef>() : nullptr;
if (!sr)
{
    continue;
}
```